### PR TITLE
Prevent copies on menu option

### PR DESCRIPTION
**Pull request by OznOg**
Prevent many copies by preventing copies and using std::move()

Closes #742

### CODE REVIEW

**Comment by OznOg** [comment]
This is almost the same PR than https://github.com/performous/performous/pull/742/ but without the use of smart pointers, mainly for comparison.

**Comment by Baklap4** [comment]
Looking at this vs #742 this looks cleaner imho

File: game/dancegraph.cc [hunk 1]
```diff
@@ -114,7 +114,7 @@ void DanceGraph::setupJoinMenu() {
 		}
 		m_selectedTrack = ConfigItem(ol); // Create a ConfigItem from the option list
 		m_selectedTrack.select(cur); // Set the selection to current level
-		m_menu.add(MenuOption("", _("Select track")).changer(m_selectedTrack)); // MenuOption that cycles the options
+		m_menu.add(MenuOption{"", _("Select track")}).changer(m_selectedTrack); // MenuOption that cycles the options
```
Comment (yoda-jm) [inline game/dancegraph.cc:117]: not a big fan of initializing with {} (especially when the members are not specified  (designators are C++20: https://en.cppreference.com/w/cpp/language/aggregate_initialization).

**Review by yoda-jm — DISMISSED** [review]

**Comment by ooshlablu** [comment]
I tested this last night with the debs and the AppImage on Ubuntu 18.04 & 20.04. Everything seems to behave normally. I didn't really notice any in-game differences which I think is supposed to be the point.


:+1:

**Review by Baklap4 — APPROVED** [review]

**Pull request merged**